kong/plugins/http-log/handler.lua

```diff
@@ -18,68 +17,146 @@ HttpLogHandler.PRIORITY = 12
 HttpLogHandler.VERSION = "1.0.0"
 
 
-local buffers = {} -- buffers per-route
+local queues = {} -- queues per-route
 
 
--- Only provide `name` when deriving from this class. Not when initializing an instance.
-function HttpLogHandler:new(name)
-  name = name or "http-log"
-  HttpLogHandler.super.new(self, name)
+-- Parse host url.
+-- @param `url` host url
+-- @return `parsed_url` a table with host details:
+-- scheme, host, port, path, query, userinfo
+local function parse_url(host_url)
+  local parsed_url = url.parse(host_url)
```

Comment:
maybe add a cache table for the parsed urls here? since its config data, they are static